src/utils/__tests__/style-formatter.spec.js

```diff
@@ -31,133 +34,145 @@ describe("style-formatter", () => {
+      expect(formattedStyle.includes("background-repeat: no-repeat")).toBe(false);
     });
 
-    it("should keep the image url but not show it when the mode changes from media to none", () => {
+    it("should keep the image url but not show it when the mode changes from media to none", async () => {
       const { backgroundImageMode } = getStyleEditorDefinition({ flags: {}, theme: {}, translator: {} }).items
         .backgroundOptions.items.backgroundImage.items;
 
       style.background.useImage = false;
       style.background.mode = "media";
       style.background.url.qStaticContentUrl = { qUrl: someUrl };
-      let formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      let formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("background-image:")).toBe(true);
 
       style.background.mode = "none";
       backgroundImageMode.change(layout);
       expect(style.background.url.qStaticContentUrl.qUrl).toEqual("/media/Logo/qlik.png");
-      formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("background-image:")).toBe(false);
 
       style.background.mode = "media";
       backgroundImageMode.change(layout);
-      formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
       expect(formattedStyle.includes("background-image:")).toBe(true);
     });
 
     it("should return specified image url and default image settings", () => {
       style.background.useImage = true;
       style.background.url.qStaticContentUrl = { qUrl: someUrl };
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
-      expect(formattedStyle.includes(`background-image: url('https://localhost${someUrl}')`)).toBe(true);
+      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app, imageUrl });
+      expect(formattedStyle.includes(`background-image: url('${imageUrl}')`)).toBe(true);
       expect(formattedStyle.includes("background-size: auto auto")).toBe(true);
       expect(formattedStyle.includes("background-position: 50% 50%")).toBe(true);
       expect(formattedStyle.includes("background-repeat: no-repeat")).toBe(true);
     });
 
-    it("should return no settings when url is missing", () => {
+    it("should return no settings when url is missing", async () => {
       style.background.useImage = true;
-      style.background.url.qStaticContentUrl = {};
-      const formattedStyle = styleFormatter.getStyles({ style, disabled, theme, app });
+      imageUrl = await new Promise((reject) => {
+        setTimeout(() => reject(undefined), 500);
+      });
+      await waitFor(() => expect(imageUrl).toBe(undefined));
```

Comment:
This is just verifying the promise that was created on the previous line.

What is that you really want to test here?

Reply:
Yes waitFor line is just for verifying the promise, resolve/reject is returned. the line after that uses imageUrl and the rest of scenario. I thought it good to have it.

Reply:
But in case `imageUrl` is always going to be `undefined`. So wrapping it in a promise does not change anything. All it does is make the test case 500 ms longer.

Can get the exact same test by doing `imageUrl = undefined`, just that is going to be a lot faster.

Reply:
True, it can only be undefined without any timeout. I've already verified that imgeUrl waits for loading anyway no need for reject case. I'll fix it.